File: unit-3/sprint-3/day-2/assignments/APiCalling_With_Service/app/src/main/java/com/example/api_calling_using_service/MainActivity.java
package com.example.api_calling_using_service;

import androidx.appcompat.app.AppCompatActivity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private Button mBtnFetchData;
    private EditText mEtUserId;
    private TextView mTvTitle;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        IntentFilter intentFilter = new IntentFilter("com.abc.BroadcastSender");
        registerReceiver(receiver,intentFilter);
    }

    private BroadcastReceiver receiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            mTvTitle.setText(intent.getStringExtra("Title"));
        }
    };

    private void initView() {
        mBtnFetchData = findViewById(R.id.btnFetchData);
        mEtUserId = findViewById(R.id.etId);
        mTvTitle = findViewById(R.id.tvTitle);
        mBtnFetchData.setOnClickListener(v -> {
            Intent intent = new Intent(MainActivity.this,MyService.class);
            intent.putExtra("userId",mEtUserId.getText().toString());
            startService(intent);
        });


    }


}

File: unit-3/sprint-4/day-3/you/Kotlin_Scoping_Functions/app/src/main/java/com/example/kotlin_scoping_functions/MainActivity.kt
package com.example.kotlin_scoping_functions

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        launchFragment()
    }

    private fun launchFragment() {
        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.add(R.id.flContainer,UserDetailFragment(),"userDetails").commit()

    }
}